### PR TITLE
Information on what to monitor for Redpanda

### LINKED ISSUE #140: Information on what to monitor for Redpanda
A customer asked me if we had a document similar to https://www.datadoghq.com/blog/monitoring-kafka-performance-metrics/. We currently have a page that describes our metrics, but we should provide more guidance on what metrics are essential to monitoring performance.

### PR DESCRIPTION
**Pull request by shorgi**
Fixes: 
- #140 
- #1046 
- #1014 
- #702 
- #1087 
- #1099 

Previews:
- Updated topic [Monitor Redpanda](https://deploy-preview-1059--redpanda-documentation.netlify.app/docs/manage/monitoring/)
- New/reorganized references for [Public and Internal Metrics](https://deploy-preview-1059--redpanda-documentation.netlify.app/docs/reference/monitor-metrics/)

**Review deadline: Thu Jan 26**

### CODE REVIEW

**Comment by tmgstevens** [comment]
LGTM. We're due to launch the new Grafana dashboards in the fairly short term. Do you want to combine these into one update?

**Comment by shorgi** [comment]
Thanks @tmgstevens for the review.  Where are the new Grafana dashboards? At minimum the docs can link to them. I'll create a separate issue for that, and if this PR is still open, I'll include the edits here.

**Comment by tmgstevens** [comment]
@shorgi here: https://github.com/redpanda-data/observability

File: docs/manage/monitoring.mdx [hunk 1]
```diff
@@ -1,316 +1,379 @@
...
+import Aliases from "@site/docs/shared/_aliases.mdx";
+import MetricsUsageTip from '@site/docs/shared/_metrics-usage-tip.mdx';
+
+Redpanda exports metrics through Prometheus endpoints for you to monitor system health and optimize system performance.
+
+A Redpanda node exports public metrics from the [/public_metrics](../../reference/public-metrics-reference) endpoint through the Admin API port (default: 9644).
 
-For information about the original metrics available on `:9644/metrics`, see [Internal Metrics](../../reference/internal-metrics).
+Before v22.2, a Redpanda node provided metrics only through the [/metrics](../../reference/internal-metrics-reference) endpoint through the Admin API port. While Redpanda still provides this endpoint, it includes many internal metrics that are unnecessary for a typical Redpanda user to monitor. Consequently, the `/public_metrics` endpoint was added to provide a smaller set of important metrics that can queried and ingested more quickly and inexpensively. The `/metrics` endpoint is now referred to as the 'internal metrics' endpoint, and Redpanda recommends it to be used for development, testing, and analysis.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:20]:
```suggestion
Before v22.2, a Redpanda node provided metrics only through the [/metrics](../../reference/internal-metrics-reference) endpoint through the Admin API port. While Redpanda still provides this endpoint, it includes many internal metrics that are unnecessary for a typical Redpanda user to monitor. Consequently, the `/public_metrics` endpoint was added to provide a smaller set of important metrics that can queried and ingested more quickly and inexpensively. The `/metrics` endpoint is now referred to as the 'internal metrics' endpoint, and Redpanda recommends that you use it for development, testing, and analysis.
```

File: docs/manage/monitoring.mdx [hunk 2]
```diff
@@ -1,316 +1,379 @@
...
+
+<MetricsUsageTip/>
+
+:::note
+To maximize monitoring performance by minimizing the cardinality of data, metrics are exported when their underlying features are in use, and are not exported when not in use. For example, a metric for consumer groups, [redpanda_kafka_consumer_group_committed_offset](../../reference/public-metrics-reference#redpanda_kafka_consumer_group_committed_offset), is not exported when no groups are registered.
+
+When monitoring internal metrics, consider enabling [aggregate_metrics](../reference/cluster-properties#aggregate_metrics) to reduce the cardinality of data to monitor.
+:::
+
+The rest of this topic covers the following about monitoring Redpanda metrics:
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:30]:
```suggestion
This topic covers the following about monitoring Redpanda metrics:
```

File: docs/manage/monitoring.mdx [hunk 3]
```diff
@@ -1,316 +1,379 @@
...
+- [Learn from examples in the Redpanda monitoring examples repository](#use-redpanda-monitoring-examples)
+- [Metrics and queries to monitor for system performance and health](#monitor-for-performance-and-health)
+- [References of public and internal metrics](#references) 
 
 ## Configure Prometheus
 
-[Prometheus](https://prometheus.io/) is a systems monitoring and alerting tool. It collects and stores metrics as time-series data identified by metric name and key/value pairs. Redpanda exports Prometheus metrics on `:9644/public_metrics`. To generate the configuration on an existing Prometheus instance, run:
+[Prometheus](https://prometheus.io/) is a system monitoring and alerting tool. It collects and stores metrics as time-series data identified by a metric name and key/value pairs.
+
+To configure and use Prometheus monitor Redpanda metrics, follow these steps.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:42]:
```suggestion
To configure and use Prometheus to monitor Redpanda metrics:
```

File: docs/manage/monitoring.mdx [hunk 4]
```diff
@@ -1,316 +1,379 @@
...
 
-`rpk generate prometheus-config`
+:::note
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:44]: The note interrupts the flow from the intro to the steps and the actual steps. Maybe put it before the intro?

File: docs/manage/monitoring.mdx [hunk 5]
```diff
@@ -1,316 +1,379 @@
...
+To maximize monitoring performance by minimizing the cardinality of data, metrics are exported when their underlying features are in use, and are not exported when not in use. For example, a metric for consumer groups, [redpanda_kafka_consumer_group_committed_offset](../../reference/public-metrics-reference#redpanda_kafka_consumer_group_committed_offset), is not exported when no groups are registered.
+
+When monitoring internal metrics, consider enabling [aggregate_metrics](../reference/cluster-properties#aggregate_metrics) to reduce the cardinality of data to monitor.
+:::
+
+The rest of this topic covers the following about monitoring Redpanda metrics:
+
+- [Configure Prometheus to monitor Redpanda metrics](#configure-prometheus)
+- [Generate a Grafana dashboard](#generate-grafana-dashboard)
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:33]:
```suggestion
- [Generate Grafana dashboard](#generate-grafana-dashboard)
```

Edited so the link matches the heading

File: docs/manage/monitoring.mdx [hunk 6]
```diff
@@ -1,316 +1,379 @@
...
+    --metrics-endpoint <node-addr>:9644/metrics> redpanda-dashboard.json 
+    ```
 
-To opt out of all metrics reporting, set `rpk.enable_usage_stats` to `false`:
+For details about the command, refer to the [rpk generate grafana-dashboard](../reference/rpk/rpk-generate/rpk-generate-grafana-dashboard.mdx) reference.
 
-```bash
-rpk config set rpk.enable_usage_stats false
-```
+In Grafana, import the generated JSON file to create a dashboard. Out of the box, Grafana generates panels tracking latency for 50%, 95%, and 99% (based on the maximum latency set), throughput, and error segmentation by type.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:121]: For the five items listed here, perhaps consider a list?

File: docs/manage/monitoring.mdx [hunk 7]
```diff
@@ -1,316 +1,379 @@
...
 
-## Monitor metrics
+You can use the imported dashboard to create new panels.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:123]:
```suggestion
To use the imported dashboard to create new panels:
```

File: docs/manage/monitoring.mdx [hunk 8]
```diff
@@ -1,316 +1,379 @@
...
-| redpanda_cluster_unavailable_partitions | Number of unavailable partitions in the cluster (that is, partitions that lack quorum among replicants). | none | gauge |
+Understanding the following unique aspects about Redpanda's architecture and data path will improve your performance debugging and tuning skills:
 
-#### Cluster-level queries
+- Redpanda replicates partitions across nodes in a cluster by using the [Raft consensus protocol](../../get-started/architecture#raft-consensus-algorithm), where each partition is a Raft consensus group. A message written from the Kafka API flows down to the Raft implementation layer that eventually directs it to a node to be stored. Observing metrics about the Raft layer thus can reveal the health of partitions and data flowing within Redpanda.
 
-Brokers in a cluster
+- Redpanda is designed with a [thread-per-core model](../../get-started/architecture#raft-consensus-algorithm) that it implements with the Seastar library. With each application thread pinned to a CPU core, when observing or analyzing the behavior of a specific application, monitor the relevant metrics with the label for the specific CPU core ('shard'), if available.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:151]:
```suggestion
- Redpanda is designed with a [thread-per-core model](../../get-started/architecture#raft-consensus-algorithm) that it implements with the Seastar library. With each application thread pinned to a CPU core, when observing or analyzing the behavior of a specific application, monitor the relevant metrics with the label for the specific CPU core ("shard"), if available.
```

Not "shard"?

File: docs/manage/monitoring.mdx [hunk 9]
```diff
@@ -1,316 +1,379 @@
...
-Kafka producer rate
+To monitor the percentage of disk consumed, use a formula with [redpanda_storage_disk_free_bytes](../../reference/public-metrics-reference#redpanda_storage_disk_free_bytes) and [redpanda_storage_disk_total_bytes](../../reference/public-metrics-reference#redpanda_storage_disk_total_bytes): 
 
-    sum(rate(redpanda_kafka_request_bytes_total{redpanda_request="produce"} [5m] )) by (redpanda_request)
+```
+1 - (sum(redpanda_storage_disk_free_bytes) / sum(redpanda_storage_disk_total_bytes))
+```
 
-### Infrastructure-level metrics
+Monitor also [redpanda_storage_disk_free_space_alert](../../reference/public-metrics-reference#redpanda_storage_disk_free_space_alert) for an alert when available disk space is low or degraded.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:182]:
```suggestion
Also monitor [redpanda_storage_disk_free_space_alert](../../reference/public-metrics-reference#redpanda_storage_disk_free_space_alert) for an alert when available disk space is low or degraded.
```

File: docs/manage/monitoring.mdx [hunk 10]
```diff
@@ -1,316 +1,379 @@
...
 
-For information about the original metrics available on `:9644/metrics`, see [Internal Metrics](../../reference/internal-metrics).
+Before v22.2, a Redpanda node provided metrics only through the [/metrics](../../reference/internal-metrics-reference) endpoint through the Admin API port. While Redpanda still provides this endpoint, it includes many internal metrics that are unnecessary for a typical Redpanda user to monitor. Consequently, the `/public_metrics` endpoint was added to provide a smaller set of important metrics that can queried and ingested more quickly and inexpensively. The `/metrics` endpoint is now referred to as the 'internal metrics' endpoint, and Redpanda recommends it to be used for development, testing, and analysis.
+
+<MetricsUsageTip/>
+
+:::note
+To maximize monitoring performance by minimizing the cardinality of data, metrics are exported when their underlying features are in use, and are not exported when not in use. For example, a metric for consumer groups, [redpanda_kafka_consumer_group_committed_offset](../../reference/public-metrics-reference#redpanda_kafka_consumer_group_committed_offset), is not exported when no groups are registered.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:25]: Odd that the link in the note renders gray and not red. It's more difficult to discern that it is a link.

File: docs/manage/monitoring.mdx [hunk 11]
```diff
@@ -1,316 +1,379 @@
...
+```
 
-Redpanda RPC error rate
+#### RPC rate
 
-    rate(redpanda_rpc_request_errors_total[5m])
+The rate of RPC requests handled should correlate with overall throughput. 
 
-Cloud storage errors
+Any errors (redpanda_node_status_rpcs_timed_out
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:222]: Looks like something is missing here?
Any errors (redpanda_node_status_rpcs_timed_out)

File: docs/manage/monitoring.mdx [hunk 12]
```diff
@@ -1,316 +1,379 @@
...
-    histogram_quantile(0.99, (sum(rate(redpanda_schema_registry_request_latency_seconds_bucket[5m])) by (le, provider, region, instance, namespace, pod)))
+To monitor Kafka consumer request latency, use the [redpanda_kafka_request_latency_seconds_bucket](../../reference/public-metrics-reference#redpanda_kafka_request_latency_seconds_bucket) histogram with the label `redpanda_request="consume"`. For example, create a query for the 99th percentile:
 
-Schema Registry request rate
+```
+histogram_quantile(0.99, sum(rate(redpanda_kafka_request_latency_seconds_bucket{redpanda_request="consume"}[5m])) by (le, provider, region, instance, namespace, pod))
+```
 
-    rate(redpanda_schema_registry_request_latency_seconds_count[5m]) + sum without(redpanda_status)(rate(redpanda_schema_registry_request_errors_total[5m]))
+The rate of Kafka consumer requests can be monitored with `redpanda_kafka_request_latency_seconds_count` with the `redpanda_request="consume"` label:
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:239]:
```suggestion
You can monitor the rate of Kafka consumer requests using`redpanda_kafka_request_latency_seconds_count` with the `redpanda_request="consume"` label:
```

File: docs/manage/monitoring.mdx [hunk 13]
```diff
@@ -1,316 +1,379 @@
...
-| Metric | Definition | Labels | Type
-| --- | --- | --- | --- |
-| redpanda_kafka_max_offset | Latest committed offset for the partition (that is, the offset of the last message safely persisted on most replicas). | redpanda_namespace, redpanda_topic, redpanda_partition | gauge |
-| redpanda_kafka_under_replicated_replicas | Number of under-replicated replicas for the partition (that is, replicas that are live but are not at the latest offset.) | redpanda_namespace, redpanda_topic, redpanda_partition | gauge |
+```
+histogram_quantile(0.99, sum(rate(redpanda_kafka_request_latency_seconds_bucket{redpanda_request="produce"}[5m])) by (le, provider, region, instance, namespace, pod))
+```
 
-#### Partition-level queries
+The rate of Kafka producer requests can be monitored with `redpanda_kafka_request_latency_seconds_count` with the `redpanda_request="produce"` label:
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:253]:
```suggestion
You can monitor the rate of Kafka producer requests with `redpanda_kafka_request_latency_seconds_count` with the `redpanda_request="produce"` label:
```

File: docs/manage/monitoring.mdx [hunk 14]
```diff
@@ -1,316 +1,379 @@
...
-Under-replicated replicas
+To monitor Redpanda internal RPC latency, use the [redpanda_rpc_request_latency_seconds](../../reference/public-metrics-reference#redpanda_rpc_request_latency_seconds) histogram. For example, create a query for the 99th percentile latency:   
 
-    redpanda_kafka_under_replicated_replicas > 0
+```
+histogram_quantile(0.99, (sum(rate(redpanda_rpc_request_latency_seconds_bucket[5m])) by (le, provider, region, instance, namespace, pod, redpanda_server)))
+```
 
-### Topic-level metrics
+The rate of internal RPC requests can be monitored with `redpanda_rpc_request_latency_seconds_count`:
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:267]:
```suggestion
You can monitor the rate of internal RPC requests with `redpanda_rpc_request_latency_seconds_count`:
```

Try to use active voice - better yet: Monitor the rate of internal...

File: docs/manage/monitoring.mdx [hunk 15]
```diff
@@ -1,316 +1,379 @@
...
+rate(redpanda_rpc_request_latency_seconds_count[5m])
+```
 
-#### Topic-level queries
 
-Kafka consumer rate
+### Partition health
 
-    sum by(redpanda_namespace,redpanda_topic)(rate(redpanda_kafka_request_bytes_total{redpanda_request="consume"}[5m]))
+The health of Kafka partitions often reflects the health of the broker nodes that host them. Thus, when alerts occur like for under-replicated partitions or more frequent leadership transfers, check for unresponsive or unavailable brokers.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:276]:
```suggestion
The health of Kafka partitions often reflects the health of the broker nodes that host them. Thus, when alerts occur for conditions such as under-replicated partitions or more frequent leadership transfers, check for unresponsive or unavailable brokers.
```

File: docs/manage/monitoring.mdx [hunk 16]
```diff
@@ -1,316 +1,379 @@
...
 
-Kafka producer rate
+With Redpanda's internal implementation of the Raft consensus protocol, the health of partitions is also reflected in any errors in the internal RPCs exchanged between Raft peers.
 
-    sum by(redpanda_namespace,redpanda_topic)(rate(redpanda_kafka_request_bytes_total{redpanda_request="produce"}[5m]))
+#### Leadership changes
 
-Partition count
+Stable clusters have a consistent balance of leaders across all nodes, with few to no leadership transfers between nodes.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:282]:
```suggestion
Stable clusters have a consistent balance of leaders across all nodes, with few-to-no leadership transfers between nodes.
```

File: docs/manage/monitoring.mdx [hunk 17]
```diff
@@ -1,316 +1,379 @@
...
-#### Broker-level queries
+An under-replicated partition is at higher risk of data loss. It also adds latency because messages must be replicated before being committed. To know when a partition isn't fully replicated, create an alert for the [redpanda_kafka_under_replicated_replicas](#redpanda_kafka_under_replicated_replicas) gauge when it is greater than zero:
 
-Kafka consumer request latency
+```
+redpanda_kafka_under_replicated_replicas > 0
+```
 
-    histogram_quantile(0.99, sum(rate(redpanda_kafka_request_latency_seconds_bucket{redpanda_request="consume"}[5m])) by (le, provider, region, instance, namespace, pod))
+Under replication can be caused by unresponsive brokers. When an alert on `redpanda_kafka_under_replicated_replicas` is triggered, identify the problem broker(s) and examine their log(s).
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:300]:
```suggestion
Under-replication can be caused by unresponsive brokers. When an alert on `redpanda_kafka_under_replicated_replicas` is triggered, identify the problem broker(s) and examine their log(s). 
```

File: docs/manage/monitoring.mdx [hunk 18]
```diff
@@ -1,316 +1,379 @@
...
+Under replication can be caused by unresponsive brokers. When an alert on `redpanda_kafka_under_replicated_replicas` is triggered, identify the problem broker(s) and examine their log(s). 
 
-Kafka producer request latency
+#### Leaderless partitions
 
-    histogram_quantile(0.99, sum(rate(redpanda_kafka_request_latency_seconds_bucket{redpanda_request="produce"}[5m])) by (le, provider, region, instance, namespace, pod))
+A healthy cluster has a leader for every partition. 
 
-Rate of Kafka consumer requests
+A partition without a leader cannot exchange messages with producers or consumers. To know when a partition doesn't have a leader, create an alert for the [redpanda_cluster_unavailable_partitions](../../reference/public-metrics-reference#redpanda_cluster_unavailable_partitions) gauge when it is greater than zero:
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:306]:
```suggestion
A partition without a leader cannot exchange messages with producers or consumers. To identify when a partition doesn't have a leader, create an alert for the [redpanda_cluster_unavailable_partitions](../../reference/public-metrics-reference#redpanda_cluster_unavailable_partitions) gauge when it is greater than zero: 
```

File: docs/manage/monitoring.mdx [hunk 19]
```diff
@@ -1,316 +1,379 @@
...
+```
 
-Rate of Kafka producer requests
+Leaderless partitions can be caused by unresponsive brokers. When an alert on `redpanda_cluster_unavailable_partitions` is triggered, identify the problem broker(s) and examine their log(s). 
 
-    rate(redpanda_kafka_request_latency_seconds_count{redpanda_request="produce"}[5m])
+#### Raft RPCs
 
-Rate of Kafka consumers
+Redpanda's Raft implementation exchanges periodic status RPCs between a node and its peers. The [redpanda_node_status_rpcs_timed_out](../../reference/public-metrics-reference#redpanda_node_status_rpcs_timed_out) gauge increases when a status RPC times out for a peer, which indicates a peer may be unresponsive and which may lead to problems with partition replication that Raft manages. Monitor for non-zero values of this gauge, and correlate it with any logged errors or changes partition replication.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:316]:
```suggestion
Redpanda's Raft implementation exchanges periodic status RPCs between a node and its peers. The [redpanda_node_status_rpcs_timed_out](../../reference/public-metrics-reference#redpanda_node_status_rpcs_timed_out) gauge increases when a status RPC times out for a peer, which indicates a peer may be unresponsive and may lead to problems with partition replication that Raft manages. Monitor for non-zero values of this gauge, and correlate it with any logged errors or changes partition replication.
```

File: docs/manage/monitoring.mdx [hunk 20]
```diff
@@ -1,316 +1,379 @@
...
-    sum(rate(redpanda_kafka_request_bytes_total{redpanda_request="consume"} [5m])) by (provider, region, instance, namespace, pod)
 
-Rate of Kafka producers
+### Consumers
 
-    sum(rate(redpanda_kafka_request_bytes_total{redpanda_request="produce"}[5m])) by (instance, provider, region, namespace, pod)
+#### Consumer group lag
 
-### Consumer group metrics
+When working with Kafka consumer groups, the consumer group lag&mdash;the difference between the broker's latest (max) offset and the group's last committed offset&mdash;is a performance indicator of how fresh is the data being consumed. While higher lag for archival consumers is expected, high lag for real-time consumers could indicate that the consumers are overloaded and thus may need their topics to be partitioned more, or to spread the load to more consumers.
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:323]:
```suggestion
When working with Kafka consumer groups, the consumer group lag&mdash;the difference between the broker's latest (max) offset and the group's last committed offset&mdash;is a performance indicator of how fresh the data being consumed is. While higher lag for archival consumers is expected, high lag for real-time consumers could indicate that the consumers are overloaded and thus may need their topics to be partitioned more, or to spread the load to more consumers.   
```

File: docs/manage/monitoring.mdx [hunk 21]
```diff
@@ -1,316 +1,379 @@
...
+### Services
 
-    sum by(redpanda_group)(redpanda_kafka_consumer_group_consumers)
+Monitor the health of specific Redpanda services with the following metrics. 
 
-Topics per consumer group
+#### Schema registry
 
-    sum by(redpanda_group)(redpanda_kafka_consumer_group_topics)
+Schema Registry request latency
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:337]:
```suggestion
Schema Registry request latency:
```

File: docs/manage/monitoring.mdx [hunk 22]
```diff
@@ -1,316 +1,379 @@
...
 
-Consumer group lag
+```
+histogram_quantile(0.99, (sum(rate(redpanda_schema_registry_request_latency_seconds_bucket[5m])) by (le, provider, region, instance, namespace, pod)))
+```
 
-    max by(redpanda_namespace, redpanda_topic, redpanda_partition)(redpanda_kafka_max_offset{redpanda_namespace="kafka"}) - on(redpanda_topic, redpanda_partition) group_right max by(redpanda_group, redpanda_topic, redpanda_partition)(redpanda_kafka_consumer_group_committed_offset)
+Schema Registry request rate
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:343]:
```suggestion
Schema Registry request rate:
```

File: docs/manage/monitoring.mdx [hunk 23]
```diff
@@ -1,316 +1,379 @@
...
-| Metric | Definition | Labels | Type
-| --- | --- | --- | --- |
-| redpanda_rest_proxy_request_errors_total | Total number of rest_proxy server errors. | redpanda_status ("5xx" \| "4xx" \| "3xx") | counter |
-| redpanda_rest_proxy_request_latency_seconds | Internal latency of request for rest_proxy. | none | histogram |
+```
+rate(redpanda_schema_registry_request_latency_seconds_count[5m]) + sum without(redpanda_status)(rate(redpanda_schema_registry_request_errors_total[5m]))
+```
 
-#### REST proxy queries
+Schema Registry request error rate
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:349]:
```suggestion
Schema Registry request error rate:
```

File: docs/manage/monitoring.mdx [hunk 24]
```diff
@@ -1,316 +1,379 @@
...
+
+```
+rate(redpanda_schema_registry_request_errors_total[5m])
+```
+
+#### REST proxy
 
 REST proxy request latency
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:357]:
```suggestion
REST proxy request latency:
```

File: docs/manage/monitoring.mdx [hunk 25]
```diff
@@ -1,316 +1,379 @@
...
 
-    histogram_quantile(0.99, (sum(rate(redpanda_rest_proxy_request_latency_seconds_bucket[5m])) by (le, provider, region, instance, namespace, pod)))
+```    
+histogram_quantile(0.99, (sum(rate(redpanda_rest_proxy_request_latency_seconds_bucket[5m])) by (le, provider, region, instance, namespace, pod)))
+```
 
 REST proxy request rate
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:363]:
```suggestion
REST proxy request rate:
```

File: docs/manage/monitoring.mdx [hunk 26]
```diff
@@ -1,316 +1,379 @@
...
 
-    rate(redpanda_rest_proxy_request_latency_seconds_count[5m]) + sum without(redpanda_status)(rate(redpanda_rest_proxy_request_errors_total[5m]))
+```
+rate(redpanda_rest_proxy_request_latency_seconds_count[5m]) + sum without(redpanda_status)(rate(redpanda_rest_proxy_request_errors_total[5m]))
+```
 
 REST proxy request error rate
```
Comment (Feediver1) [inline docs/manage/monitoring.mdx:369]:
```suggestion
REST proxy request error rate:
```

**Review by Feediver1 — APPROVED** [review]
See suggestions.

Thanks for your work on this @shorgi - very important and helpful to our customers.

**Comment by shorgi** [comment]
@rkruze @VladLazar - Please review/approve this new PR for the updated Monitoring topic. I plan to merge it end of tomorrow if there are no outstanding comments. 

@tmgstevens - the new Grafana dashboards are linked from here: https://deploy-preview-1059--redpanda-documentation.netlify.app/docs/manage/monitoring/#use-redpanda-monitoring-examples

File: docs/manage/monitoring.mdx [hunk 27]
```diff
@@ -1,316 +1,372 @@
...
+When working with Kafka consumer groups, the consumer group lag&mdash;the difference between the broker's latest (max) offset and the group's last committed offset&mdash;is a performance indicator of how fresh the data being consumed is. While higher lag for archival consumers is expected, high lag for real-time consumers could indicate that the consumers are overloaded and thus may need their topics to be partitioned more, or to spread the load to more consumers.   
 
-    sum(rate(redpanda_kafka_request_bytes_total{redpanda_request="consume"} [5m])) by (provider, region, instance, namespace, pod)
+To monitor consumer group lag, create a query with the [redpanda_kafka_max_offset](../../reference/public-metrics-reference#redpanda_kafka_max_offset) and [redpanda_kafka_consumer_group_committed_offset](redpanda_kafka_consumer_group_committed_offset) gauges:
 
-Rate of Kafka producers
+```
+max by(redpanda_namespace, redpanda_topic, redpanda_partition)(redpanda_kafka_max_offset{redpanda_namespace="kafka"}) - on(redpanda_topic, redpanda_partition) group_right max by(redpanda_group, redpanda_topic, redpanda_partition)(redpanda_kafka_consumer_group_committed_offset)
```
Comment (VladLazar) [inline docs/manage/monitoring.mdx:321]: @tmgstevens I recall you were looking into consumer lag metrics. Does this look right to you?

**Review by VladLazar — COMMENTED** [review]
Nice change in my opinion. Please re-check the hyperlinks. I found a few that were 404s or pointing to the wrong place.

File: docs/manage/monitoring.mdx [hunk 28]
```diff
@@ -1,316 +1,372 @@
...
-    redpanda_cluster_topics
+#### CPU usage
 
-Partitions in a cluster
+For the total CPU uptime, monitor [redpanda_uptime_seconds_total](../../reference/public-metrics-reference#redpanda_uptime_seconds_total) and [redpanda_cpu_busy_seconds_total](../../reference/public-metrics-reference#redpanda_cpu_busy_seconds_total):
 
-    redpanda_cluster_partitions
+```
+rate(redpanda_uptime_seconds_total[5m])
+rate(redpanda_cpu_busy_seconds_total[5m])
```
[the remark this answers is not in the document]
Comment (shorgi) [reply docs/manage/monitoring.mdx:163]: @VladLazar - `redpanda_cpu_busy_seconds_total` itself is the total busy (non-idle) time, right? So you're suggesting querying busy_time / uptime to get the percentage in use?

[the remark this answers is not in the document]
Comment (VladLazar) [reply docs/manage/monitoring.mdx:163]: Not quite. `redpanda_cpu_busy_seconds_total` is a monotonic counter that tracks how many seconds the core that a specific shard is pinned to was busy for. The rate of change (or derivative) tells you what percentage of that shard was in use at a given point in time.

File: docs/manage/monitoring.mdx (hunk 27 above)
Comment (shorgi) [reply docs/manage/monitoring.mdx:326]: @tmgstevens I'll be merging this PR by EOB tomorrow, Tue Jan 31, please review here or if needed resolve it in a subsequent issue.

**Comment by shorgi** [comment]
> I've just noticed that some public metrics slipped through and are not available on the [reference page](https://docs.redpanda.com/docs/beta/reference/public-metrics-reference/). For instance, `cloud_client_backoff`, `cloud_client_upload_backoff` and `cloud_client_download_backoff` are not there, but can be found [here](https://docs.redpanda.com/docs/beta/manage/monitoring/#monitor-metrics).

@VladLazar my apologies for not seeing your comment earlier. Creating a new PR to add these missing metrics.

**Pull request merged**